src/js/extensions/ccf/node.cpp

```diff
@@ -69,21 +78,7 @@ namespace ccf::js
           ctx, "Passed %d arguments but expected two", argc);
       }
 
-      auto gov_effects = static_cast<ccf::AbstractGovernanceEffects*>(
-        JS_GetOpaque(this_val, node_class_id));
-
-      if (gov_effects == nullptr)
-      {
-        return JS_ThrowInternalError(ctx, "Node state is not set");
-      }
-
-      auto tx_ptr = jsctx.globals.tx;
-
-      if (tx_ptr == nullptr)
-      {
-        return JS_ThrowInternalError(
-          ctx, "No transaction available to open service");
-      }
+      LOAD_FROM_EXTENSION(jsctx);
```

Comment:
A bit unfortunate to have a macro that creates variables in the current scope, I think three assignments with a JS_CHECK using the right message would have been slightly longer but a little less magic.

Reply:
Agreed - I'll do the copy-paste. I thought I was going to have to repeat this pattern for every extension, and looking to reduce boiler-plate, but actually it's only needed by a few of these functions so fits with the existing boilerplate

Reply:
I would mildly prefer that, but this is already a big PR, so also fine to keep this way.